### DISCUSSION TITLE
Loki endpoint

### DISCUSSION THREAD

**Discussion by hijackit** [tkowalcz/tjahzi#44] [Q&A] [answered]
Hi, I'm trying to use this cool library to send logs to a cloud-based loki instance using log4j2

I expected to see a configuration field like the following:
`<url>http://host:port/loki/api/v1/push</url>`
instead of:
`<host>loki-host</host>
<port>loki-port</port>`

How does this library send messages to loki? Through the gRPC protocol? 
I suppose that the port should match the `grpc_listen_port` loki configuration, is that right?
Any chance to send messages to the loki rest API? 

Thanks!

**Comment by tkowalcz** [comment, marked as answer]
Hi, thanks for kind words. Sorry, I should have documented that better.  Is it Grafana cloud you are connecting to? I wanted to try it out myself and document it, I'll make that a priority. 

Now back to your question - tjahzi does http POST to `/loki/api/v1/push`. So instead of
```xml
<url>http://host:port/loki/api/v1/push</url>
```

you can do:
```
<host>host</host>
<port>port</port>
```

and it should work all the same. Please let me know if I can help or modify something on my end.

**Reply by hijackit** [reply]
It make sense! 
I gave it a try, but I think there is another problem now :)
The loki endpoint is behind HTTPS, so what I did is:
```
<host>host</host>
<port>443</port>
```
but I think that the http client used by tjahzi does not recognize that the 443 port is "special" and does not start a correct https connection. 

Setting the log4j configuration to verbose, I can see in the logs the response of our reverse-proxy (nginx):
`400 The plain HTTP request was sent to HTTPS port`

Any suggestion?
Thanks for your help, much appreciated!

**Reply by tkowalcz** [reply]
I think that at this point the only option is to add a missing feature :) I have created ticket #45 to track this. I should be able to implement it today, worst case - end of this week.

**Reply by hijackit** [reply]
It would be amazing, thanks for your help!

**Comment by tkowalcz** [comment]
There is a new release with support for https and basic auth: https://github.com/tkowalcz/tjahzi/releases/tag/tjahzi-parent-0.9.6.

I have added test case to the CI that integrates with Grafana Cloud. Also run a logger for several hours that delivered logs there. Seems to work fine :)

**Reply by hijackit** [reply]
I confirm that it works like a charm also in our environment!
